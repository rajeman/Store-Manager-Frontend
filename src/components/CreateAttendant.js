import React from 'react';
import { connect } from 'react-redux';
import { fetchUser } from '../actions/auth';
import Header from './Header';
import Footer from './Footer';
import AttendantCreateForm from './AttendantCreateForm';
import NotFoundPage from './NotFoundPage';

class CreateAttendant extends React.Component {
  componentDidMount() {
    this.props.dispatch(fetchUser());
  }
  render() {
    const { level } = this.props.auth.userDetails;
    if (level === 1) {
      return (
        <NotFoundPage />
      )
    }
    return level === 2 ? (
      <div id="cover">
        <Header />
        <AttendantCreateForm />
        <Footer />
      </div>
    ) : <div></div>

  }
}

const mapStateToProps = (state) => state;

export default connect(mapStateToProps)(CreateAttendant);

