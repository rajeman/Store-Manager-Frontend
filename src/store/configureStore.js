import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import productReducer from '../reducers/products';
import authReducer from '../reducers/auth';
import saleReducer from '../reducers/records';
import navigationReducer from '../reducers/navigation';
import attendantReducer from '../reducers/attendants';
import cartReducer from '../reducers/cart';

export default () => createStore(
  combineReducers({
    products: productReducer,
    auth: authReducer,
    navigation: navigationReducer,
    sales: saleReducer,
    attendants: attendantReducer,
    cart: cartReducer,
  }), applyMiddleware(thunk),
);
