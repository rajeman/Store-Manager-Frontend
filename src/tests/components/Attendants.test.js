import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import axios from 'axios';
import CreateAttendant from '../../components/CreateAttendant';

jest.mock('axios');
jest.mock('../../helpers/cart');

describe('ATTENDANT CREATE TEST SUITE', () => {
  it('should render the form for creating attendant', () => {
    const initialState = {
      products: {
        productCreate: 'STATE_CREATE_FAILED',
        productCreateError: ''
      },
      auth: {
        userDetails: {
          level: 2
        }
      },
      attendants: {
        attendantCreateState: 'STATE_CREATING'
      }
    };
    const response = {
        data:
        {
          message: ''
        }
      };
    axios.get.mockResolvedValue(response);
    const mockStore = configureStore([thunk]);
    const store = mockStore(initialState);
    const component = mount(
      <Provider store={store}>
        <MemoryRouter>
          <CreateAttendant />
        </MemoryRouter>
      </Provider>
    );
    expect(component.exists()).toBe(true);
  });

  it('should dispatch the create attendant action when attendant account is being created', () => {
    const initialState = {
      products: {
        productCreate: '',
        productCreateError: ''
      },
      auth: {
        userDetails: {
          level: 2
        }
      },
      attendants: {
        attendantCreateState: 'STATE_CREATING'
      }
    };
    const response = {
      data:
      {
        message: 'successfully created attendant'
      }
    };
    axios.post.mockImplementation(() => Promise.resolve(response));
    const mockStore = configureStore([thunk]);
    const store = mockStore(initialState);
    const component = mount(
      <Provider store={store}>
        <MemoryRouter>
          <CreateAttendant />
        </MemoryRouter>
      </Provider>
    );
    component.find('form').simulate('submit', {});
    const dispatchedActions = store.getActions();
    expect(dispatchedActions[0].state).toEqual('STATE_CREATING');
  });

  it('should dispatch the \'create attendant error\' if create fails', () => {
    const initialState = {
      products: {
        productCreate: 'STATE_CREATE_FAILED',
        productCreateError: ''
      },
      auth: {
        userDetails: {
          level: 2
        }
      },
      attendants: {
        attendantCreateState: 'STATE_CREATING'
      }
    };
    const response = {
      data:
      {
        message: 'successfully created attendant'
      }
    };
    axios.post.mockImplementation(() => Promise.reject(response));
    const mockStore = configureStore([thunk]);
    const store = mockStore(initialState);
    const component = mount(
      <Provider store={store}>
        <MemoryRouter>
          <CreateAttendant />
        </MemoryRouter>
      </Provider>
    );
    component.find('form').simulate('submit', {});
    const dispatchedActions = store.getActions();
    expect(dispatchedActions[0].state).toEqual('STATE_CREATING');
  });
});

