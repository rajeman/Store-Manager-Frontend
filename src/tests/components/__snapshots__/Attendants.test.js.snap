// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ATTENDANT CREATE TEST SUITE should render the form for creating attendant 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "m29cvr",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "m29cvr",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(CreateAttendant)>
        <CreateAttendant
          attendants={
            Object {
              "attendantCreateState": "STATE_CREATING",
            }
          }
          auth={
            Object {
              "userDetails": Object {
                "level": 2,
              },
            }
          }
          dispatch={[Function]}
          products={
            Object {
              "productCreate": "STATE_CREATE_FAILED",
              "productCreateError": "",
            }
          }
        >
          <div
            id="cover"
          >
            <Header>
              <header>
                <div
                  className="container"
                >
                  <div
                    id="website-logo"
                  >
                    <h1>
                      Store Manager
                    </h1>
                  </div>
                  <nav>
                    <ul>
                      <li>
                         
                        <NavLink
                          activeClassName="active"
                          ariaCurrent="true"
                          to="/dashboard/products"
                        >
                          <Route
                            path="/dashboard/products"
                          >
                            <Link
                              aria-current={false}
                              replace={false}
                              to="/dashboard/products"
                            >
                              <a
                                aria-current={false}
                                href="/dashboard/products"
                                onClick={[Function]}
                              >
                                Home
                              </a>
                            </Link>
                          </Route>
                        </NavLink>
                      </li>
                      <li>
                        <a
                          href="#"
                        >
                          About
                        </a>
                      </li>
                      <li>
                        <a
                          href="#"
                        >
                          Services
                        </a>
                      </li>
                      <Connect(Signout)>
                        <Signout
                          attendants={
                            Object {
                              "attendantCreateState": "STATE_CREATING",
                            }
                          }
                          auth={
                            Object {
                              "userDetails": Object {
                                "level": 2,
                              },
                            }
                          }
                          dispatch={[Function]}
                          products={
                            Object {
                              "productCreate": "STATE_CREATE_FAILED",
                              "productCreateError": "",
                            }
                          }
                        >
                          <li>
                            <a
                              href=""
                              onClick={[Function]}
                            >
                              Signout
                            </a>
                          </li>
                        </Signout>
                      </Connect(Signout)>
                    </ul>
                  </nav>
                </div>
              </header>
            </Header>
            <Connect(AttendantCreateForm)>
              <AttendantCreateForm
                attendants={
                  Object {
                    "attendantCreateState": "STATE_CREATING",
                  }
                }
                auth={
                  Object {
                    "userDetails": Object {
                      "level": 2,
                    },
                  }
                }
                dispatch={[Function]}
                products={
                  Object {
                    "productCreate": "STATE_CREATE_FAILED",
                    "productCreateError": "",
                  }
                }
              >
                <div
                  className="cont"
                >
                  <section>
                    <div
                      id="user-form"
                    >
                      <div
                        className="container"
                      >
                        <form
                          onSubmit={[Function]}
                        >
                          <h3>
                            New Attendant
                          </h3>
                          <div>
                            <label>
                              Name:
                            </label>
                            <br />
                            <input
                              id="name"
                              name="aname"
                              pattern=".{3,}"
                              placeholder="Enter Attendant's Name"
                              required={true}
                              title="Name must be at least 3 characters"
                              type="text"
                            />
                          </div>
                          <div>
                            <label>
                              Email:
                            </label>
                            <br />
                            <input
                              id="email"
                              name="aemail"
                              placeholder="Enter Attendant's Email"
                              required={true}
                              type="email"
                            />
                          </div>
                          <div
                            className="not-registered"
                          >
                            <span
                              className="lds-hourglass"
                            />
                          </div>
                          <button
                            type="submit"
                          >
                            Create
                          </button>
                          <br />
                        </form>
                      </div>
                    </div>
                  </section>
                </div>
              </AttendantCreateForm>
            </Connect(AttendantCreateForm)>
            <Footer>
              <footer>
                <p>
                   Store Manager Copyright Â© 2018
                </p>
              </footer>
            </Footer>
          </div>
        </CreateAttendant>
      </Connect(CreateAttendant)>
    </Router>
  </MemoryRouter>
</Provider>
`;
